### ISSUE TITLE
Can't pass serviceaccountname from WorkflowTemplates on web submit

### ISSUE THREAD

**Issue by dmayle** [argoproj/argo#3138]
Checklist:
 
**What happened**:
I'm converting a working Template to a WorkflowTemplate.  The serviceAccountName field is properly set on the WorkflowTemplate, but when I use the web UI to submit the template, I have to manually copy the serviceAccountName field to the input box, or the workflow will be submitted using the default serviceaccount

**What you expected to happen**:
I expect the serviceaccountname from the WorkflowTemplate to be copied to the Template if it's set on the WorkflowTemplate

**How to reproduce it (as minimally and precisely as possible)**:
```
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: repro
spec:
  entrypoint: hello
  serviceAccountName: alpine-repo
  templates:
  - name: hello
    script:
      image: alpine:3.10
      command: [sh]
      source: |
        echo hello
```

**Anything else we need to know?**:

**Environment**:

- Argo version:
```
$ argo version
argo: v2.8.0
  BuildDate: 2020-05-11T22:55:16Z
  GitCommit: 8f696174746ed01b9bf1941ad03da62d312df641
  GitTreeState: clean
  GitTag: v2.8.0
  GoVersion: go1.13.4
  Compiler: gc
  Platform: linux/amd64
```
- Kubernetes version :
```
$ kubectl version -o yaml
clientVersion:
  buildDate: "2020-04-16T11:56:40Z"
  compiler: gc
  gitCommit: 52c56ce7a8272c798dbc29846288d7cd9fbae032
  gitTreeState: clean
  gitVersion: v1.18.2
  goVersion: go1.13.9
  major: "1"
  minor: "18"
  platform: linux/amd64
serverVersion:
  buildDate: "2020-05-20T12:43:34Z"
  compiler: gc
  gitCommit: 2e7996e3e2712684bc73f0dec0200d64eec7fe40
  gitTreeState: clean
  gitVersion: v1.18.3
  goVersion: go1.13.9
  major: "1"
  minor: "18"
  platform: linux/amd64

```

---

**Message from the maintainers**:

If you are impacted by this bug please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by sarabala1979** [comment]
This issue fixed using `workflowtemplateRef` to refer to the `WorkflowTemplate`.
Fixed:  https://github.com/argoproj/argo/pull/3116.

**Comment by sarabala1979** [comment]
Fixed: #3116
